app/master/build_artifact.py

```diff
@@ -177,3 +189,23 @@ def _artifact_directory(build_id, subjob_id=None, atom_id=None, result_root=None
             return os.path.join(result_root, str(build_id), BuildArtifact.ATOM_DIR_FORMAT.format(subjob_id, atom_id))
         else:
             raise ValueError('Specified one of either subjob_id or atom_id. Must either specify both or neither.')
+
+    @staticmethod
+    def _subjob_and_atom_ids(directory_name):
+        """
+        Infer the subjob and atom id's from an artifact directory name. This name should be of format:
+
+        "artifact_SubjobId_AtomId"
+
+        The method raises a ValueError if the directory_name does not match expected convention.
+
+        :type directory_name: str
+        :return: A tuple, with the first element being the subjob id, and the second element being the atom id.
+        :rtype: (int, int)
+        """
+        id_match = re.search(r'artifact_(\d*)_(\d*)$', directory_name)
```

Comment:
I think 'artifact_0_' might pass this with an empty string match, but it would still raise a value error when converting to int below so it prob doesn't matter too much.